packages/firestore/src/core/view.ts

```diff
@@ -82,6 +90,18 @@ export class View {
   private limboDocuments = documentKeySet();
   /** Document Keys that have local changes */
   private mutatedKeys = documentKeySet();
+  /**
+   * The first snapshot version that Watch has sent us during the lifetime of
+   * of this view. Any document edited after will have `hasPendingWrites` set
+   * until Watch catches up.
+   */
+  private firstSnapshotVersion = SnapshotVersion.MIN;
```

Comment:
I removed these snapshot versions:
- Since we store UnknownDocuments in the database, we no longer need to check whether the snapshots "went back in time".
- I added a special `computeInitialChanges()` method that gets rid of the need for `firstSnapshotVersion`. This method differs from computeDocChanges in that ignores `hasCommittedMutations` for the first view snapshot.